src/js.c

```diff
@@ -3041,30 +3030,22 @@ js_PullSubscribeAsync(natsSubscription **newsub, jsCtx *js, const char *subject,
     // Initialize fetch parameters.
     if (s == NATS_OK)
     {
-        fetch->startTimeMilli = nats_Now();
-        fetch->lifetime = *lifetime;
-
-        if (fetch->lifetime.Batch == 0)
-            fetch->lifetime.Batch = INT_MAX;
-
-        fetch->nextf = _autoNextFetchRequest;
-        fetch->nextClosure = (void *)fetch;
-        if (jsOpts != NULL)
-        {
-            fetch->fetchSize = jsOpts->PullSubscribeAsync.FetchSize;
-            fetch->keepAhead = jsOpts->PullSubscribeAsync.KeepAhead;
-            fetch->completeCB = jsOpts->PullSubscribeAsync.CompleteHandler;
-            fetch->completeCBClosure = jsOpts->PullSubscribeAsync.CompleteHandlerClosure;
-
-            if (jsOpts->PullSubscribeAsync.NextHandler != NULL)
-            {
-                fetch->nextf = jsOpts->PullSubscribeAsync.NextHandler;
-                fetch->nextClosure = jsOpts->PullSubscribeAsync.NextHandlerClosure;
-            }
-        }
+        fetch->status = NATS_OK;
+        fetch->startTimeMillis = nats_Now();
 
-        if (fetch->fetchSize == 0)
-            fetch->fetchSize = NATS_DEFAULT_ASYNC_FETCH_SIZE;
+#define _set(_f, _v, _nil, _def) fetch->_f = ((jsOpts != NULL) && (jsOpts->PullSubscribeAsync._v != _nil)) ? jsOpts->PullSubscribeAsync._v : _def
```

Comment:
At this point, I wonder why not embed a PullSubscribeAsync option block inside the jsFetch and copy the user provided values (if jsOpts != NULL) into it?

Reply:
ok.